### PR TITLE
New free functor

### PR DESCRIPTION
**Pull request by maxsnew**
(Draft because I still need to clean some things up)

Give a new definition for the free functor between freely generated categories. Has the following differences from previous:
1. Based on quivers rather than graphs (better definitional behavior in some uses)
2. Freely builds in the functor rather than using a graph homomorphism
3. Defined using new Presented category construction
4. UMP is specified as an eliminator into a displayed category

Because of (2) this seems to avoid the need for any `Eq` manipulation in the definition of the eliminator. Need to port the solver to this to make sure the definitional behavior is right but on inspection it looks good.

### CODE REVIEW

**Comment by maxsnew** [comment]
Ended up adding/changing a few more things, copying from the recent commit: 

1. Move Quiver to Data.Quiver, define Graph^D over Quiver, reformulate free category data in terms of Graph^D
2. Reformulate Presented Cat to take in a category rather than quiver
3. Use named module in presented cat

File: Cubical/Categories/Constructions/Free/Functor/AltPresented.agda [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+          -- F⟪ id A ⟫ ≡ id (F ⟅ A ⟆)
+          satisfies-axioms (inl A) =
+            cong (𝓕 .F-homᴰ) (s .F-id)
+            ◁ 𝓕 .F-idᴰ
+          -- F⟪ f ⋆ g ⟫ ≡ F⟪ f ⟫ ⋆ F⟪ g ⟫
+          satisfies-axioms (inr (_ , _ , _ , f , g)) =
+            cong (𝓕 .F-homᴰ) (s .F-seq _ _)
+            ◁ 𝓕 .F-seqᴰ _ _
+
+module CoUnit {C : Category ℓc ℓc'} {D : Category ℓd ℓd'} (F : Functor C D)
```
Comment (stschaef) [inline Cubical/Categories/Constructions/Free/Functor/AltPresented.agda:122]: counit makes this sound like an adjunction, but I don't see what functors go into that adjunction. 

Is this just augmenting D by freely adding in the image of F, or is this also doing some adjunction business. Like also adding in the image of `FG d`, where `F : C -> D`, `G : D -> C` are adjoint.

Comment (stschaef) [reply Cubical/Categories/Constructions/Free/Functor/AltPresented.agda:122]: like a sort of free syntax to represent an adjoint to a given functor F. idk maybe I'm reading too much into this. I guess I just don't see if this is the free category you mention at the top of file, or if its an instance

File: Cubical/Categories/Displayed/Base/More.agda [hunk 2]
```diff
@@ -61,3 +62,23 @@ module _ {C : Category ℓC ℓC'} {D : Category ℓD ℓD'}
...
+  private
+    module C = Category C
+    module D = Category D
+
+  open Categoryᴰ Dᴰ
+  open Functor F
+  open Functorᴰ
+
+  -- TODO: better name?
+  reindexΠ : Functorᴰ F (reindex Dᴰ F) Dᴰ
```
Comment (stschaef) [inline Cubical/Categories/Displayed/Base/More.agda:80]: maybe "reindex" with a superscript "d" for displays? or maybe "reindexing-functor"?

File: Cubical/Categories/Displayed/Section.agda [hunk 3]
```diff
@@ -102,6 +103,7 @@ module _ (C : Category ℓC ℓC') (P : Preorderᴰ C ℓD ℓD') where
     Section≡ _ (λ c → refl) (λ f → refl)
   PreorderSectionIsoCatSection .leftInv a = refl
 
+-- | TODO: this should go somewhere else
 module _ (C : Category ℓC ℓC') (D : Category ℓD ℓD') where
   weaken : Categoryᴰ C ℓD ℓD'
```
Comment (stschaef) [inline Cubical/Categories/Displayed/Section.agda:108]: why "weaken"

**Review by stschaef — COMMENTED** [review]
Other than the couple naming questions above, this all looks good. 

Although, what cubical version are you on. On cubical HEAD, I seem to be getting issues loading these file stemming from something in `NaturalTransformation.Properties` or something

**Comment by maxsnew** [comment]
> Although, what cubical version are you on. On cubical HEAD, I seem to be getting issues loading these file stemming from something in `NaturalTransformation.Properties` or something

You need to upgrade your version of Agda

File: Cubical/Categories/Constructions/Free/Functor/AltPresented.agda (hunk 1 above)
Comment (maxsnew) [reply Cubical/Categories/Constructions/Free/Functor/AltPresented.agda:122]: This is the co-unit of a comonad on the 2-category of functors. I.e., for every functor F : C -> D we get a functor Free C - FreeFunctor > Free(C,D,F) and there is a square from this to the original functor. This gets used in the solver. I did use the name counit at least in a comment in the free category module (https://github.com/agda/cubical/blob/master/Cubical/Categories/Constructions/Free/Category/Base.agda#L159)

File: Cubical/Categories/Displayed/Section.agda (hunk 3 above)
Comment (maxsnew) [reply Cubical/Categories/Displayed/Section.agda:108]: The idea is to think of a displayed category `C` over a category `G` like a dependent type `G |- C`. Then we can think of an ordinary category as dependent on an empty context `* |- C` and so this operation takes `* |- D` and turns it into `C |- wk(D)`, i.e., weakening.

File: Cubical/Categories/Displayed/Base/More.agda (hunk 2 above)
Comment (maxsnew) [reply Cubical/Categories/Displayed/Base/More.agda:80]: Idk I think `reindexΠ` is more informative than both of those. In terms of the total space of the displayed categories involved, this is a projection.

Comment (stschaef) [reply Cubical/Categories/Displayed/Base/More.agda:80]: It seems opposite to some of the cubical naming scheme though. Like, `isSetΠ` and other similarly named things are refering to `Π` types

Comment (maxsnew) [reply Cubical/Categories/Displayed/Base/More.agda:80]: then maybe just lower-case pi?

Comment (maxsnew) [reply Cubical/Categories/Displayed/Base/More.agda:80]: I went with "forgetReindex" since it's a kind of forgetful functor

**Pull request merged**